### PR TITLE
Update "Using the Okteto CLI" page and add new "Overview" page to manifest section

### PR DESCRIPTION
**Pull request by omnomagonz**
This PR contains the following changes:

- Updated content for the `Core concepts` > `Using the Okteto CLI` page for (hopefully) clarity and improved information flow
- Removed the verbose section about the Okteto Manifest to `Okteto manifest` > `Overview` to serve as a baseline page for educating users about the manifest

### CODE REVIEW

File: src/content/manifest/overview-manifest.mdx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      - 9229:9229
+    sync:
+      - .:/usr/src/app
+```
+
+This example shows how to [build](reference/okteto-manifest.mdx/#build-object-optional) the container image of the **api** microservice, how to [deploy](reference/okteto-manifest.mdx/#deploy-object-optional) the application using its helm chart,
+and how to [develop](reference/okteto-manifest.mdx/#dev-object-optional) on the **api** microservice by synchronizing the local filesystem on the container path `/usr/src/app`.
+If you want to learn more about what all you can configure in this manifest, head over to the [Okteto Manifest Reference](reference/okteto-manifest.mdx).
+
+### Built-in Environment Variables
```
Comment (pchico83) [inline src/content/manifest/overview-manifest.mdx:33]: this is dup in both content, is it on purpose?

**Review by pchico83 — APPROVED** [review]

File: src/content/manifest/overview-manifest.mdx (hunk 1 above)
Comment (omnomagonz) [reply src/content/manifest/overview-manifest.mdx]: Nope, not on purpose and I think fixed now? Or at least I'm not seeing the duplicate content anymore

**Comment by omnomagonz** [comment]
@pchico83 I made a larger edit to this PR. I removed a lot of content from this `Overview` page that I plan to move to a different page based on the outline you had suggested in Slack.

Would you mind re-reviewing this to make sure there's nothing egregiously wrong? 🙏 

cc @codyjlandstrom for additional review if you have a moment

File: src/content/core/using-okteto-cli.mdx [hunk 2]
```diff
@@ -5,143 +5,64 @@ sidebar_label: Using the Okteto CLI
...
-    wait: true
-deploy:
-  - echo ${OKTETO_DEPENDENCY_DATABASE_VARIABLE_USERNAME}
-  - echo ${OKTETO_DEPENDENCY_DATABASE_VARIABLE_PASSWORD}
-```
-
-Note that `USERNAME` has an static value, and `PASSWORD` is dynamically generated.
+In the Okteto manifest you can define the [`dev` section](reference/okteto-manifest.mdx#dev-object-optional) which specifically informs Okteto about how to build and deploy a [development container](deploy/development-environments.mdx) for each service or component you'd like to work on. This feature enables you to make local changes to your deployed application and see those changes reflected immediately within the remote development container in Okteto. The Okteto CLI creates a bridge between your local machine and your Okteto development container so you can continue coding in your favorite IDE and see changes live on the deployed version of your application as soon as you hit save.
 
 ### Built-in Tools When Deploying To Okteto
```
Comment (pchico83) [inline src/content/core/using-okteto-cli.mdx:63]: I don't think this section belongs here. This are tools available in your `deploy` commands

File: src/content/manifest/overview-manifest.mdx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+| [destroy](#destroy) | Configure a list of commands to destroy all resources created in the `deploy` and `external` sections. | [Learn more](reference/okteto-manifest.mdx#destroy-string-optional) |
+| [dev](#dev) | Configure a list of development containers and their behavior to define how `okteto up` works when you start working on a specific, configured service in your development environment. | [Learn more](reference/okteto-manifest.mdx#dev-object-optional) |
+| [external](#external) | Configure a list of external resources used by your application's development environment. These are resources that exist outside of the Okteto cluster (e.g. cloud resources, dashboards). | [Learn more](reference/okteto-manifest.mdx#external-object-optional). |
+
+There are other sections to help you build your ideal development experience in Okteto, but this document focuses solely on introducing you to the core concepts of the Okteto manifest. You can learn more about the other Okteto manifest options in the [manifest reference](reference/okteto-manifest.mdx).
+
+## Build
```
Comment (pchico83) [inline src/content/manifest/overview-manifest.mdx:25]: I would have a dedicated doc for each section, as we already do for `Variables` and `External Resources`. I like the quick introduction here, but we should add a link to the detailed page when it's available

File: src/content/manifest/overview-manifest.mdx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+This section contains the instructions for Okteto to build the images for each service in your application. All services defined in this section will appear in the Okteto UI as part of a single development environment that represents the whole of your application. Each individual service will appear as its own development container that’s part of the development environment for your application.
```
Comment (pchico83) [inline src/content/manifest/overview-manifest.mdx:27]: The services in the UI depend on the resources created by the `deploy` section. `build` doesn't have a representation in our UI today

File: src/content/manifest/overview-manifest.mdx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    depends_on: base
+    args:
+      SOURCE_IMAGE: ${OKTETO_BUILD_BASE_IMAGE}
+    secrets:
+      npmrc: .npmrc
+```
+
+This configuration defines images to be built for three services: `base`, `api`, and `frontend`. It also defines a `context` for each image, which tells Okteto which folder/subfolder to use for building each container image. For example: `default (.)` tells Okteto to use the current folder for the `base` image, the `api` subfolder for the `api` image, and the `frontend` subfolder for the `frontend` image.
+
+You can learn more about these additional parameters [here](reference/okteto-manifest.mdx).
```
Comment (pchico83) [inline src/content/manifest/overview-manifest.mdx:50]: Should we link to the `build` reference?

File: src/content/manifest/overview-manifest.mdx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    command: helm upgrade --install postgresql postgresql/postgresql-11.6.21.tgz -f postgresql/values.yml --version 11.6.21
+    - name: Deploy Frontend
+    command: helm upgrade --install frontend frontend/chart --set image=${OKTETO_BUILD_FRONTEND_IMAGE}
+    - name: Deploy API
+    command: helm upgrade --install api api/chart --set image=${OKTETO_BUILD_API_IMAGE} --set load=${API_LOAD_DATA:-true}
+```
+
+In this section the manifest instructs Okteto to execute three commands: deploy a postgresql database, deploy the `frontend` image (using an environment variable), and deploy the `api` image (also using environment variables).
+
+These commands are executed within the development environment created by Okteto enabling you to create a production-like environment for developing your application.
```
Comment (pchico83) [inline src/content/manifest/overview-manifest.mdx:72]: Should we add a link to the `deploy` reference docs?

File: src/content/manifest/overview-manifest.mdx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+```yaml
+destroy:
+  image: okteto/pipeline-runner:1.0.0-sam
+  commands:
+    - name: destroy worker service
+      command: |
+        sam delete --no-prompts --stack-name "${OKTETO_NAMESPACE}-voting-texkhnclxd" --region us-east-1
+```
+
+In this example `destroy` section we are deleting the worker service running on the image container specified in the `image:` attribute. When the development environment is deleted, Okteto will execute this command ensuring orphaned resources don't remain active.
```
Comment (pchico83) [inline src/content/manifest/overview-manifest.mdx:91]: Should we add a link to the `deploy` reference docs?

File: src/content/manifest/overview-manifest.mdx [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    - name: Try it out!
+      url: https://github.com/okteto/voting-app-with-external-resources
+  lambda:
+    icon: aws
+    notes: docs/lambda.md
+    endpoints:
+    - name: function
+```
+
+This section provides only the metadata to represent and link to external resources in the Okteto UI and does not create any resources in a cloud provider. Creating or destroying the resources themselves is done in the [deploy](#deploy) and [destroy](#destroy) sections, respectively.
```
Comment (pchico83) [inline src/content/manifest/overview-manifest.mdx:114]: Should we add a link to the `external` reference docs?

File: src/content/manifest/overview-manifest.mdx [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+      - 9229:9229
+    sync:
+      - api:/usr/src/app
+  frontend:
+    command: yarn start
+    sync:
+      - frontend:/usr/src/app
+```
+
+In this example, the `api` service you define is *separate* from the original `api` service created in the `build` and `deploy` sections. This `api` service in the `dev` section will tell Okteto to create a new development container that exists specifically to sync code between your local computer and the remote Okteto development container so you can see your changes in real-time within a production-like environment. This also enables you to share an endpoint for others to view and collaborate on your work with you without having to do any extra work.
```
Comment (pchico83) [inline src/content/manifest/overview-manifest.mdx:137]: Should we add a link to the `dev` reference docs?

File: src/content/core/using-okteto-cli.mdx (hunk 2 above)
Comment (omnomagonz) [reply src/content/core/using-okteto-cli.mdx:63]: @pchico83 I'm not sure I understand. I wanted to convey here that "dev containers exist and are how we provide part of the Okteto experience".

I'm not sure what "tools available in your `deploy` commands" means in this context.

If this section were re-written would it be a better fit or do you think we should remove it entirely?

File: src/content/manifest/overview-manifest.mdx (hunk 3 above)
Comment (omnomagonz) [reply src/content/manifest/overview-manifest.mdx:25]: Agreed. That's the goal. So is the feedback here basically "Let's make sure to circle back and add a link to the more detailed page once it's available"?

File: src/content/manifest/overview-manifest.mdx (hunk 6 above)
Comment (omnomagonz) [reply src/content/manifest/overview-manifest.mdx:72]:
```suggestion
These commands are executed within the development environment created by Okteto enabling you to create a production-like environment for developing your application.

You can learn more about this section [here](reference/okteto-manifest.mdx#deploy-string-optional).
```

File: src/content/manifest/overview-manifest.mdx (hunk 8 above)
Comment (omnomagonz) [reply src/content/manifest/overview-manifest.mdx:116]:
```suggestion
This section provides only the metadata to represent and link to external resources in the Okteto UI and does not create any resources in a cloud provider. Creating or destroying the resources themselves is done in the [deploy](#deploy) and [destroy](#destroy) sections, respectively.

You can learn more about this section [here](reference/okteto-manifest.mdx#external-object-optional).
```

File: src/content/manifest/overview-manifest.mdx (hunk 9 above)
Comment (omnomagonz) [reply src/content/manifest/overview-manifest.mdx:141]:
```suggestion
In this example, the `api` service you define is *separate* from the original `api` service created in the `build` and `deploy` sections. This `api` service in the `dev` section will tell Okteto to create a new development container that exists specifically to sync code between your local computer and the remote Okteto development container so you can see your changes in real-time within a production-like environment. This also enables you to share an endpoint for others to view and collaborate on your work with you without having to do any extra work.

You can learn more about this section [here](reference/okteto-manifest.mdx#dev-object-optional).
```

File: src/content/manifest/overview-manifest.mdx (hunk 7 above)
Comment (omnomagonz) [reply src/content/manifest/overview-manifest.mdx:93]:
```suggestion
In this example `destroy` section we are deleting the worker service running on the image container specified in the `image:` attribute. When the development environment is deleted, Okteto will execute this command ensuring orphaned resources don't remain active.

You can learn more about this section [here](reference/okteto-manifest.mdx#deploy-string-optional).
```

File: src/content/core/using-okteto-cli.mdx (hunk 2 above)
Comment (omnomagonz) [reply src/content/core/using-okteto-cli.mdx:63]: Removing the section for now and will add it back in here or wherever we deem it appropriate in a follow-on PR 👍

File: src/content/manifest/overview-manifest.mdx (hunk 4 above)
Comment (omnomagonz) [reply src/content/manifest/overview-manifest.mdx]: Addressed in the latest commit: https://github.com/okteto/docs/pull/573/commits/8c12641cdaa13f8e009f004e6d30f536bf24849b

Moved this section to the `Deploy` section

**Review by pchico83 — APPROVED** [review]

**Pull request merged**